src/script/interpreter.cpp

```diff
@@ -495,6 +495,7 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                     if (vfExec.empty())
                         return set_error(serror, SCRIPT_ERR_UNBALANCED_CONDITIONAL);
                     vfExec.back() = !vfExec.back();
+                    fExec = fExec ? false : !count(vfExec.begin(), vfExec.end(), false);
```

Comment:
in `OP_ELSE`, if `fExec` is true, the original `vfExec` must be by definition all true. Since `OP_ELSE` will flip to last `vfExec` item, the resulting `vfExec` must have exactly one false. So we know `fExec` must become false.

If `fExec` is false, there is at least one false in `vfExec` but we don't know where is it, so we have to count as usual

Reply:
`std::count`?

Reply:
Could improve by using `std::find` since the actual count is not necessary?